src/Components/Body/Chat/Chat.tsx

```diff
@@ -0,0 +1,44 @@
+import ChatLeftFold from "./LeftFold";
+import ChatRightFold from "./RightFold";
+import "./style.css";
+import { useState } from "react";
+import {useAllUsers} from '../../../hooks/useAllUsers'
+import {Spinner} from '../../Spinner/Spinner'
+import { ErrorState } from "../../Error/ErrorState";
+import { ChatMessageType, ChatListDataType } from "../../../Types/types";
+export function Chat() {
+  const {allUserDetails: chatList,loading,error, updateQuery} = useAllUsers()
+  const [selectedChat, setSelectedChat] = useState<number>(0);
+
+  if(loading) return < Spinner color='#000000' size={100} />
+  if(error)  return < ErrorState />
+
+  function handleNewMessage(newMessageObj: ChatMessageType){
```

Comment:
use useCallback.

Note - As the codebase grows we do not exactly know whether the callback which we are getting as a prop to the component is memoized or not. This might cause performance implications. So, most of time prefer to use useCallback.

For instances where we are passing callbacks directly to native elements we can avoid using useCallback.

https://kentcdodds.com/blog/usememo-and-usecallback